confd/conf.d/opennms-datasources.xml.toml

```diff
@@ -0,0 +1,19 @@
+[template]
+src = "opennms-datasources.xml.tpl"
+dest = "/opt/opennms/etc/opennms-datasources.xml"
+keys = [
+    "/opennms/database/connection-pool/factory",
+    "/opennms/database/connection-pool/idletimeout",
+    "/opennms/database/connection-pool/logintimeout",
+    "/opennms/database/connection-pool/minpool",
+    "/opennms/database/connection-pool/maxpool",
+    "/opennms/database/connection-pool/maxsize",
```

Comment:
Can you provide an example for the environment variable that I have to set when using the character `-` as part of a name ?

I've tried `OPENNMS_DATABASE_CONNECTION_POOL_MAXSIZE` and it didn't work; I believe because that might be translated to `/opennms/database/connection/pool/maxsize` (btw, this format would fix the issue).

I haven't tried `OPENNMS_DATABASE_CONNECTION-POOL_MAXSIZE`, as that would be an invalid name for an environment variable (as `-` has a special meaning).

Reply:
Due to https://github.com/kelseyhightower/confd/issues/737 or https://github.com/kelseyhightower/confd/issues/531, I believe we don't have many options when using the `env` backend. In other words, the way the current implementation on this PR is laid out will work with consul, etcd, zookeeper, redis, etc., but not with environment variables.

In certain cases, replace `-` with `/` makes sense like the above example, but there are other places where it would make it harder for external systems.

For example, change the key `/opennms/trapd/new-suspect-on-trap` to `/opennms/trapd/new/suspect/on/trap`, in order to define the environment variable like `OPENNMS_TRAPD_NEW_SUSPECT_ON_TRAP`; otherwise, to be consistent with all the backends we would have to write the environment variable like `OPENNMS_TRAPD_NEWSUSPECTONTRAP` which makes my brain hurt.

Reply:
I opted to make all the environment variables readable (i.e. define the keys like `/opennms/trapd/new/suspect/on/trap`). Maybe that might not look natural when using other backends, but in theory should work. I'm open for suggestions here, as maybe it worth discard `env` and use `file` as the backend.